common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -226,11 +226,11 @@ def _clear_bulk_ops_record(self, course_key):
                 course_key.replace(org=None, course=None, run=None, branch=None)
             ]
 
-    def _start_outermost_bulk_operation(self, bulk_write_record, course_key):
+    def _start_outermost_bulk_operation(self, bulk_write_record, course_key, ignore_case=False):
         """
         Begin a bulk write operation on course_key.
         """
-        bulk_write_record.initial_index = self.db_connection.get_course_index(course_key)
+        bulk_write_record.initial_index = self.db_connection.get_course_index(course_key, ignore_case=ignore_case)
         # Ensure that any edits to the index don't pollute the initial_index
         bulk_write_record.index = copy.deepcopy(bulk_write_record.initial_index)
         bulk_write_record.course_key = course_key
```

Comment:
I'm concerned that `bulk_write_record` didn't change. In particular, consider this case:

    with bulk_operations(CourseLocator('org', 'course', 'run'), ignore_case=True):
        lower_case = modulestore().create_course(CourseLocator('org', 'course', 'run'), ignore_case=True)
        upper_case = modulestore().get_course(CourseLocator('ORG', 'COURSE', 'RUN'), ignore_case=True)

        # Does upper_case throw an error?